src/sets.jl

```diff
@@ -220,17 +222,19 @@ See [here](http://lpsolve.sourceforge.net/5.5/SOS.htm) for a description of SOS
 constraints and their potential uses.
 This is a shortcut for the `MathOptInterface.SOS2` set.
 """
-struct SOS2 <: AbstractVectorSet
-    weights::Vector{Float64}
-    function SOS2(weights::AbstractVector = Float64[])
-        return new(convert(Vector{Float64}, weights))
+struct SOS2{T} <: AbstractVectorSet
+    weights::Vector{T}
+    function SOS2{T}(weights::AbstractVector = T[]) where {T}
+        return new{T}(convert(Vector{T}, weights))
     end
 end
-function moi_set(set::SOS2, dim::Int)
+SOS2() = SOS2{Int}()
```

Comment:
I think this is breaking. Because solvers support `MOI.SOS2{Float64}`, not `MOI.SOS2{Int}`.

Reply:
This is going to get converted in `model_convert` in the macro